security/subsystem/src/main/java/org/jboss/as/security/SecurityDomainResourceDefinition.java

```diff
@@ -63,9 +64,12 @@
  */
 class SecurityDomainResourceDefinition extends SimpleResourceDefinition {
 
+    static final String INFINISPAN_CACHE_TYPE = "infinispan";
+    static final RuntimeCapability<Void> INFINISPAN = RuntimeCapability.Builder.of("org.wildfly.security.domain.infinispan", true).build();
```

Comment:
This is ok if this is recorded as a private capability. Probably this should just be "org.wildfly.security.legacy-security-domain" or some name like that with service type SecurityDomainContext. Nothing special about infinispan in the capability name, and then the capability description in wildfly-capabilities records the infinispan cache capability as an optional requirement. But I'm not sure SecurityDomainContext is a good contract. For example it exposes public setters.